### ISSUE TITLE
Photos lose order when given new date/times when fixing colliding date/times

### ISSUE THREAD

**Issue by joshkendrick** [31TenSoftware/RenameMediaTool#1]
There's currently an issue with the way new date/times are determined when multiple photos have the same date/time (the for loop that's about here https://github.com/31TenSoftware/RenameMediaTool/blob/master/src/com/thirtyonetensoftware/renamemediatool/ProcessWorker.java#L214).

If a couple pictures have colliding date/times, currently we loop to get all the photos with that date/time, then loop through those, and add x seconds to each photo. If the seconds added is high enough (say like 5 seconds), then it's possible a new date/time is picked that is later than a photo with a valid date/time.

For example, 3 photos:
1) 2015-10-20 5:49:25pm
2) 2015-10-20 5:49:25pm
3) 2015-10-20 5:39:27pm

Currently, #2 will be given a new date time
1) 2015-10-20 5:49:25pm
2) 2015-10-20 5:49:30pm
3) 2015-10-20 5:39:27pm

--> which makes it so they'll be reordered 1, 3, 2 -- when the order should be preserved (1, 2, 3).

The solution might be to implement some rank or order that can be checked and all subsequent photos' times bumped back when something like this happened (so in this case #3's time would become 2015-10-20 5:39:31pm)

**Comment by joshkendrick** [comment]
I thought of 3 "solutions":
1. The rank thing mentioned above. This has the drawback of modifying the date/time of files whose date/times were accurate before I changed them.
2. Modify the naming scheme so files with the same date/time have the same count, but then add another count at the end i.e. 2015-10-20_0003_01, 2015-10-20_0003_02. The drawback here is it creates the illusion that these files are somehow connected even though they arent; it makes it look like they're different versions of the same picture kind of.. It makes it look like files are named because of time, but really it's order.
3. Lower the time added to the lowest possible amount (1 second). The drawback is this doesn't **really** fix the issue, but makes it less likely to occur.
   Note: it is still possible that changing the date/times here in 3 still causes photos to become out of order.

I went with #3. The majority of the cases where date/time are the exact same, I **do** want to stagger the times because they were photos that were scanned in and I set the date/time manually to some date at noon. The drawback of 3 was the most appealing of the negatives.

The alternative was not change the date/time at all, but in the cases where I scanned a bunch of files in and gave them all the same date/time, I want the date/times to be staggered.
